e2e/cypress/integration/messaging/long_draft_spec.js

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Messaging', () => {
+    before(() => {
+        // # Make sure the viewport is the expected one, so written lines always create new lines
+        cy.viewport(1000, 660);
+
+        // # Login and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M18699-Leave a long draft in the main input box', () => {
+        const lines = ['Lorem ipsum dolor sit amet,',
+            'consectetur adipiscing elit.',
+            'Nulla ac consectetur quam.',
+            'Phasellus libero lorem,',
+            'facilisis in purus sed, auctor.'];
+
+        cy.visit('/ad-1/channels/town-square');
+
+        // # Get the height before starting to write
+        cy.get('#post_textbox').should('be.visible').clear().then((post) => {
+            cy.wrap(parseInt(post[0].clientHeight, 10)).as('initialHeight').as('previousHeight');
+        });
+
+        // # Post first line to use
+        cy.get('#post_textbox').type(lines[0]);
+
+        // # For each line
+        for (let i = 1; i < lines.length; i++) {
+            // # Post the line
+            cy.get('#post_textbox').type('{shift}{enter}').type(lines[i]);
+
+            cy.get('#post_textbox').invoke('attr', 'height').then((height) => {
+                // * Previous height should be lower than the current heigh
+                cy.get('@previousHeight').should('be.lessThan', parseInt(height, 10));
+
+                // # Store the current height as the previous height for the next loop
+                cy.wrap(parseInt(height, 10)).as('previousHeight');
+            });
+        }
+
+        // # Visit a different channel and verify textbox
+        cy.get('#sidebarItem_off-topic').click({force: true});
```

Comment:
nit: On initial test data generation (by `make test-data`) , I'm seeing that "user-1" is not a member of off-topic channel.  With that, above test will fail in case no preceding test that makes user-1  as member of it.  One way is to switch the steps - first by visiting URL (line 68-74) then by clicking on the sidebar.

Reply:
On d9beb8e I did the visiting on beforeall, since can be considered a setup step.

Hope it makes sense!

Reply:
Thanks, that works too!